File: app/src/main/java/com/slavik/aproximadorfunciones/modelo/Punto.java
package com.slavik.aproximadorfunciones.modelo;

public class Punto {
    public double x, y;
    public boolean visible;

    public Punto(double x, double y){
        this.x = x;
        this.y = y;
        visible = true;
    }

    public Punto(double x, double y, boolean visible) {
        this.x = x;
        this.y = y;
        this.visible = visible;
    }
}

File: app/src/main/java/com/slavik/aproximadorfunciones/mmc/dominio/modelo/Punto.kt
package com.slavik.aproximadorfunciones.mmc.dominio.modelo

import com.slavik.aproximadorfunciones.mmc.util.Formato

data class Punto(
    var x : Double,
    var y : Double,
    var visible: Boolean = true
){
    override fun toString(): String {
        return "(${Formato.decimal(x)}; ${Formato.decimal(y)})"
    }

    constructor(x: Float, y: Float, visible: Boolean = true) : this(x.toDouble(), y.toDouble(), visible)
    constructor(x: Int, y: Int, visible: Boolean = true) : this(x.toDouble(), y.toDouble(), visible)
}